### PR TITLE
Usability updates

### PR DESCRIPTION
**Pull request by nuxeh**
- print version number on --version flag
- expand `~` in config and database paths
- add `systemd` unit
- add Debian packaging
- add manual page

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,24 +96,33 @@ If history is enabled, no database type is specified in the `[database]`
...
+## Running as a service
+
+### Debian package install
+
+If you install using the Debian package, this also installs a systemd unit for
+running as a service, e.g.:
 
-## Auto-start
+    systemctl enable url-bot-rs.service
+    systemctl start url-bot-rs.service
```
Comment (polyzen) [inline README.md]: This can be shortened to `systemctl enable --now url-bot-rs.service`

Comment (polyzen) [reply README.md]: Or even without `.service`, but that may be unwanted.

File: systemd/url-bot-rs.service [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=url-bot-rs
+After=network.target
+Requires=network.target
+
+[Service]
```
Comment (polyzen) [inline systemd/url-bot-rs.service]: Probably want to enforce the use of a `url-bot-rs` user, and perhaps even the hardening flags commented out here: 
https://limnoria.readthedocs.io/en/latest/use/supybot-botchk.html#systemd-service

File: README.md (hunk 1 above)
Comment (nuxeh) [reply README.md]: Nice, i'll update the readme.

Another thought, might it be worth giving instructions for manual install of the unit, e.g.
```
sudo cp systemd/url-bot-rs.service /lib/systemd/system/
sudo chmod 644 /lib/systemd/system/url-bot-rs.service
```
or even:
```
sudo install -m 644 systemd/url-bot-rs.service /lib/systemd/system/
```

File: systemd/url-bot-rs.service (hunk 2 above)
Comment (nuxeh) [reply systemd/url-bot-rs.service]: I've added running as `url-bot-rs` user in 394f883, and an issue for adding hardening options as issue #89

Comment (nuxeh) [reply systemd/url-bot-rs.service]: DynamicUser seems dodgy from what i've read, a hard-coded user seems more reliable.

File: README.md [hunk 3]
```diff
@@ -113,16 +113,52 @@ After this, the bot may be started manually by running `url-bot-rs`.
...
+directory files are deleted, according to Debian packaging guidelines. This
+keeps UIDs more deterministic, and allows re-installation or upgrade without
+losing the bot's configuration.
+
+### Manual systemd install
+
+To set up systemd manually, the unit file must be copied, and the `url-bot-rs`
+user must be created on the system. From inside the project repository:
+
+    sudo install -m 644 systemd/url-bot-rs.service /lib/systemd/system/
```
Comment (polyzen) [inline README.md]: Manual unit installation should use `/etc/systemd/system/`:
https://www.freedesktop.org/software/systemd/man/systemd.unit.html#Unit%20File%20Load%20Path

File: debian/postinst [hunk 4]
```diff
@@ -0,0 +1,2 @@
+useradd -m --system url-bot-rs --shell /usr/sbin/nologin
+systemctl enable url-bot-rs.service
```
Comment (polyzen) [inline debian/postinst]: `enable --now`?

File: README.md [hunk 5]
```diff
@@ -113,16 +113,52 @@ After this, the bot may be started manually by running `url-bot-rs`.
...
+    sudo useradd -m --system url-bot-rs --shell /usr/sbin/nologin
+    systemctl enable --now url-bot-rs.service
```
Comment (polyzen) [inline README.md]: Missing `sudo` here and elsewhere for consistency.

**Pull request merged**